apps/src/templates/teacherDashboard/TeacherDashboard.jsx

```diff
@@ -84,6 +87,10 @@ function TeacherDashboard({
     }
   });
 
+  const teacherNavV2Enabled =
```

Comment:
I considered putting this in a useEffect / useState but I actually didn't see any other examples of this currently in our codebase.  I would be interested in your decision tree here.  In theory, we really only should need this to be calculated once on load...

Reply:
I would agree, we only need to do it once. We could use `useEffect` or we can use `_.once()` ([docs](https://lodash.com/docs/4.17.15#once))

Reply:
I am adding a useEffect, however I am still a little unclear as to why this is actually better.  useState and useEffect feels less clean to me and less readable.  Researching it, I don't see a strong case for performance improvements.  I'd love 2 sentences more on why you think useState/useEffect is a better move here.